webapp/src/features/adminAddRsu/AdminAddRsu.js

```diff
@@ -0,0 +1,262 @@
+  const selectedSnmpGroup = useSelector(selectSelectedSnmpGroup);
+  const organizations = useSelector(selectOrganizations);
+  const selectedOrganizations = useSelector(selectSelectedOrganizations);
+  const submitAttempt = useSelector(selectSubmitAttempt);
+
+  const {
+    register,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm();
+
+  useEffect(() => {
+    dispatch(getRsuCreationData());
+  }, []);
+
+  return (
+    <div>
+      <Form onSubmit={handleSubmit((data) => dispatch(submitForm({ data, reset })))}>
+        <Form.Group className="mb-3" controlId="ip">
+          <Form.Label>RSU IP</Form.Label>
+          <Form.Control
+            type="text"
+            placeholder="Enter RSU IP"
+            {...register("ip", {
+              required: "Please enter the RSU's IP address",
+              pattern: {
+                value: /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/,
+                message: "Please enter a valid IP address",
+              },
+            })}
+          />
+          {errors.ip && <p className="errorMsg">{errors.ip.message}</p>}
+        </Form.Group>
+
+        <Form.Group className="mb-3" controlId="latitude">
+          <Form.Label>Latitude</Form.Label>
+          <Form.Control
+            type="text"
+            placeholder="Enter RSU Latitude"
+            {...register("latitude", {
+              required: "Please enter the RSU latitude",
+              pattern: {
+                value: /^(\+|-)?(?:90(?:(?:\.0{1,8})?)|(?:[0-9]|[1-8][0-9])(?:(?:\.[0-9]{1,8})?))$/,
+                message: "Please enter a valid latitude",
+              },
+            })}
+          />
+          {errors.latitude && <p className="errorMsg">{errors.latitude.message}</p>}
+        </Form.Group>
+
+        <Form.Group className="mb-3" controlId="longitude">
+          <Form.Label>Longitude</Form.Label>
+          <Form.Control
+            type="text"
+            placeholder="Enter RSU Longitude"
+            {...register("longitude", {
+              required: "Please enter the RSU longitude",
+              pattern: {
+                value: /^(\+|-)?(?:180(?:(?:\.0{1,8})?)|(?:[0-9]|[1-9][0-9]|1[0-7][0-9])(?:(?:\.[0-9]{1,8})?))$/,
+                message: "Please enter a valid longitude",
+              },
+            })}
+          />
+          {errors.longitude && <p className="errorMsg">{errors.longitude.message}</p>}
+        </Form.Group>
+
+        <Form.Group className="mb-3" controlId="milepost">
+          <Form.Label>Milepost</Form.Label>
+          <Form.Control
+            type="text"
+            placeholder="Enter RSU Milepost"
+            {...register("milepost", {
+              required: "Please enter the RSU milepost",
+              pattern: {
+                value: /^\d*\.?\d*$/,
```

Comment:
The first time I attempted to add an RSU with milepost 17 this failed, but subsequent attempts seem to be succeeding - any idea what would cause that? It clearly displayed the "please enter a valid number" message, but now I can't get it to reproduce.

Reply:
If you enter "17." without any decimal places, the regex currently marks that as invalid, could that have been what happened? From my testing it seems to be working fine for all valid inputs